src/settings.js

```diff
@@ -1235,9 +1235,14 @@ var IN_TEST_HARNESS = 0;
 // If true, enables support for pthreads.
 var USE_PTHREADS = 0;
 
-// Specifies the number of web workers that are preallocated before runtime is
-// initialized. If 0, workers are created on demand.
+// PTHREAD_POOL_SIZE specifies the number of web workers that are created
+// before the main runtime is initialized. If 0, workers are created on
+// demand. If PTHREAD_POOL_ONLY_PREWARM = 0, then the workers will be fully
+// loaded (available for use) prior to the main runtime being initialized. If
+// PTHREAD_POOL_ONLY_PREWARM = 1, then the workers will only be created and
+// have their runtimes loaded on demand after the main runtime is initialized.
 var PTHREAD_POOL_SIZE = 0;
+var PTHREAD_POOL_ONLY_PREWARM = 0;
```

Comment:
I think we can maybe improve this flag a little. One issue is that we call it `prewarm` here but `preallocated` in the implementation. It would be better to make that uniform. How about `preallocate` in both places?

A second thing is `ONLY_PREWARM` seems less clear than `PREWARM`.

Overall, what do you think about having this be `var PTHREAD_POOL_PREALLOCATE = 1;`?

Reply:
Done